SimTKcommon/include/SimTKcommon/internal/ClonePtr.h

```diff
@@ -202,7 +207,11 @@ template <class T> class ClonePtr {
         other.reset(p);
         reset(otherp);
     }
-     
+
+    DEPRECATED_14("use get() instead")
+    const T* getPtr()  const  { return get(); }
+    DEPRECATED_14("use upd() instead")
+    T* updPtr() { return upd(); }
```

Comment:
> If they are already using it, they don't need the documentation.

This is not true when inheriting someone else's code (e.g., going through Jack/Tim's predictive code).

Freeglut is deprecated on macs. Say it were deprecated everywhere in the world, but I still wanted to add the antialiasing and thus needed the documentation on the deprecated methods. (This is a little contrived, since the antialiasing isn't specific to freeglut).

> If they are thinking about using it, they definitely should not do so, so again we shouldn't help them with documentation.

My opinion is we should help them with documentation that says they shouldn't use the method.

The reality of deprecating is that users may have lots of code that depends on deprecated stuff and it may be a difficult task to move to the new methods. They may be on a tight research deadline (well, in this case they shouldn't have moved to the next version of a dependency), and might need to use the deprecated methods for a while and thus would want the documentation.

Having documentation could be helpful throughout the process of moving to the non-deprecated methods. Especially for more complicated changes, like OpenSim moving from `virtual void connectToModel()` to `virtual void extendConnectToModel()`, which presents a subtle opportunity to introduce bugs (if the user forgets to rename `Super::connectToModel()` to `Super::extendConnectToModel()`). I think the doxygen deprecated warning could be very strongly worded. In cases where it's not simply "use this other method instead", the documentation might be necessary to understand how the functionality changed between versions.